src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -483,43 +483,41 @@ export class ResultsViewPageStore {
     readonly genePanelInformation = remoteData<GenePanelInformation>({
         await:()=>[
             this.studyToMutationMolecularProfile,
-            this.studyToDataQueryFilter,
             this.genes,
             this.samples,
-            this.patients,
-            this.studyIds
+            this.patients
         ],
         invoke:async()=>{
-            const studies = this.studyIds.result!;
-            const results:GenePanelData[] = _.flatten(await Promise.all(studies.map(studyId=>{
-                const mutationMolecularProfile = this.studyToMutationMolecularProfile.result![studyId];
-                const dataQueryFilter = this.studyToDataQueryFilter.result![studyId];
-                if (mutationMolecularProfile && dataQueryFilter) {
-                    return client.getGenePanelDataUsingPOST({
-                        molecularProfileId: mutationMolecularProfile.molecularProfileId,
-                        genePanelDataFilter:{
-                            entrezGeneIds: this.genes.result!.map(x=>x.entrezGeneId),
-                            sampleIds: dataQueryFilter.sampleIds,
-                            sampleListId: dataQueryFilter.sampleListId
-                        } as GenePanelDataFilter
+            const studyToMutationMolecularProfile = this.studyToMutationMolecularProfile.result!;
+            const sampleMolecularIdentifiers:SampleMolecularIdentifier[] = [];
+            this.samples.result!.forEach(sample=>{
+                const profile = studyToMutationMolecularProfile[sample.studyId];
+                if (profile) {
```

Comment:
provide for case where there's no profile

Reply:
if theres no profile, we simply dont add that sample to the query

Reply:
but if sampleMolecularIdentifiers doesn't accumulate anything (as would happen with CNA only query?) then we'll call service with empty array?

Reply:
oh oops i forgot to push that update